core/src/main/java/team/unnamed/gui/core/GUIListeners.java

```diff
@@ -55,13 +64,23 @@ public void onClose(InventoryCloseEvent event) {
             guiData
                     .getCloseAction()
                     .ifPresent(action -> action.accept(event));
+
+            Player player = (Player) humanEntity;
+            plugin.getServer().getScheduler().runTaskLater(plugin, () -> {
+                for(ItemStack item : player.getInventory().getContents()) {
+                    if(item != null && ItemStackNBTFactory.getInstance().hasTag(item, "UNNAMEDGUI")) {
```

Comment:
`ItemStackNBTFactory#getInstance()` should be called only once since it uses reflection and it is also not necessary to do it all the time